lib/matrix_client.ex

```diff
@@ -146,11 +166,51 @@ defmodule MatrixClient do
     Session.get_invites(pid)
   end
 
+  def rooms(pid) do
+    Session.get_rooms(pid)
+  end
+
   def accept_invite(pid, room_id) do
     %{status: 200} = join_room(pid, room_id)
     Session.delete_invite(pid, room_id)
   end
 
+  def next_room_messages(pid, room_id) do
+    get_room_messages(pid, room_id, "f")
+  end
+
+  def prev_room_messages(pid, room_id) do
+    get_room_messages(pid, room_id, "b")
+  end
+
+  defp get_room_messages(pid, room_id, direction) do
+    {:ok, url} = Session.get(pid, :url)
+    {:ok, token} = Session.get(pid, :token)
+    aliases = Session.get_aliases(pid)
+
+    new_id =
```

Comment:
As an aside, I'm seeing the following block a lot—refactor opportunity in future.